gearcatalog/frontend/src/components/layout/Header.tsx

```diff
@@ -0,0 +1,85 @@
+import React from "react";
+import { Link } from "react-router-dom";
+import { connect } from "react-redux";
+import { logout } from "../../actions/auth";
+import { RootReducerState } from "../../reducers/reducer.types";
+import { User } from "../../actions/actions.types";
+
+type HeaderProps = {
+  logout: (token: string) => void,
+  isAuthenticated: boolean,
+  token: string,
+  user: User | null,
+};
+
+const Header = ({ logout, isAuthenticated, token, user }: HeaderProps) => {
+  const authLinks = (
+    <ul className="navbar-nav ml-auto mt-2 mt-lg-0">
+      <span className="navbar-text text-light mr-3">
+        <strong>
+          {user ? `Welcome ${user.username[0].toUpperCase() +
+          user.username.slice(1)}` : ""}
```

Comment:
What if I want my username to be all lowercase!?

Reply:
>.>;;